### DISCUSSION TITLE
Keyball46 firmware v1.0.0 RC2

### DISCUSSION THREAD

**Discussion by koron** [Yowkees/keyball#39] [Show and tell]
これよりも新しいファームが https://github.com/Yowkees/keyball/discussions/46 に置いてあります。

---

Keyball46 の v1.0.0 のリリース候補(RC2)のビルド済みファームウェアを用意しました。

種類 | 右ボール | 両ボール | 左ボール | 説明
---|---|---|---|---
**VIA対応** | [keyball_via_1.0.0-rc.2.zip](https://github.com/Yowkees/keyball/files/6727237/keyball_via_1.0.0-rc.2.zip) | [keyball_via_Both_1.0.0-rc.2.zip](https://github.com/Yowkees/keyball/files/6727238/keyball_via_Both_1.0.0-rc.2.zip) | [keyball_via_Left_1.0.0-rc.2.zip](https://github.com/Yowkees/keyball/files/6727239/keyball_via_Left_1.0.0-rc.2.zip) | VIA (Remap) 対応
**組み立てテスト** | [keyball_test_1.0.0-rc.2.zip](https://github.com/Yowkees/keyball/files/6727241/keyball_test_1.0.0-rc.2.zip) |[keyball_test_Both_1.0.0-rc.2.zip](https://github.com/Yowkees/keyball/files/6727242/keyball_test_Both_1.0.0-rc.2.zip) | [keyball_test_Left_1.0.0-rc.2.zip](https://github.com/Yowkees/keyball/files/6727243/keyball_test_Left_1.0.0-rc.2.zip) | テスト用キーマップ<br>RGB LEDが赤→緑→青で点灯
**ミニマル** | [keyball_default_1.0.0-rc.2.zip](https://github.com/Yowkees/keyball/files/6727240/keyball_default_1.0.0-rc.2.zip) | N/A | N/A | OLED&RGB LEDに非対応<br>USBはボール側に接続

このバージョンのファームウェアではトラックボールの有無に応じて基板ごとに書き込むファームウェアが異なります。各ZIPには `_rev1_ball_` と `_rev1_noball_` の2つのファームが含まれています。`_rev1_ball_` をトラックボールのある方に、`_rev1_noball_` をトラックボールのない方に書き込んでください。

例: VIA対応右ボール版の中身

* `keyball_rev1_ball_via.hex` - トラックボールのあるPCB用
* `keyball_rev1_noball_via.hex` - トラックボールのないPCB用

## ファームウェアの選び方

通常の利用時には **VIA対応** 版を使えば十分です。

組み立てる時のテストには **組み立てテスト** 版が便利です。RGB LEDの動作確認に適した点灯パターンと、各キーのダイオードの確認に適したキーマップが含まれています。

**ミニマル** は参考資料用です。ご自身でソースからファームをビルドする際のテンプレートとなる `default` の動作確認ができます。

**Comment by Yowkees** [comment]
表で見れば一目瞭然でずね。それにしてもすごい！ビルドガイドでの説明方法も考えたいと思います。

**Comment by Yowkees** [comment]
OLED表示ばっちりです！めちゃくちゃ面白いです。
キー表示も入りきりましたか！すごい！
